temporian/core/serialize.py

```diff
@@ -0,0 +1,340 @@
+            "to an unexpected input signature. "
+            f"Expected: {input_args.keys()} Effective: {op.inputs().keys()}"
+        )
+
+    if op.outputs().keys() != output_args.keys():
+        raise ValueError(
+            f"Restoring the operator {src.operator_def_key} lead "
+            "to an unexpected output signature. "
+            f"Expected: {output_args.keys()} Effective: {op.outputs().keys()}"
+        )
+
+    if op.attributes().keys() != attribute_args.keys():
+        raise ValueError(
+            f"Restoring the operator {src.operator_def_key} lead to an"
+            " unexpected attributes signature. Expected:"
+            f" {attribute_args.keys()} Effective: {op.attributes().keys()}"
+        )
+    # TODO: Deep check of equality of the inputs / outputs / attributes
+
+    # Override the inputs / outputs / attributes
+    op.set_inputs(input_args)
+    op.set_outputs(output_args)
+    op.set_attributes(attribute_args)
+    return op
+
+
+def _serialize_event(src: Event) -> pb.Event:
+    return pb.Event(
+        id=_identifier(src),
+        sampling_id=_identifier(src.sampling()),
+        feature_ids=[_identifier(f) for f in src.features()],
+    )
+
+
+def _unserialize_event(
+    src: pb.Event, samplings: Dict[str, Sampling], features: Dict[str, Feature]
+) -> Event:
+    if src.sampling_id not in samplings:
+        raise ValueError(f"Non existing sampling {src.sampling_id}")
+
+    def get_feature(key):
+        if key not in features:
+            raise ValueError(f"Non existing feature {key}")
+        return features[key]
+
+    return Event(
+        sampling=samplings[src.sampling_id],
+        features=[get_feature(f) for f in src.feature_ids],
+    )
+
+
+def _serialize_feature(
+    src: Feature, input_features: Set[Feature]
+) -> pb.Feature:
+    return pb.Feature(
+        id=_identifier(src),
+        name=src.name(),
+        dtype=_type_to_proto(src.dtype()),
+        sampling_id=_identifier(src.sampling()),
+        creator_event_id=(
+            _identifier(src.creator()) if src not in input_features else None
+        ),
+    )
+
+
+def _unserialize_feature(
+    src: pb.Feature, samplings: Dict[str, Sampling]
+) -> Feature:
+    if src.sampling_id not in samplings:
+        raise ValueError(f"Non existing sampling {src.sampling_id}")
+
+    return Feature(
+        name=src.name,
+        dtype=_type_from_proto(src.dtype),
+        sampling=samplings[src.sampling_id],
+        creator=None,
+    )
+
+
+def _serialize_sampling(
+    src: Sampling, input_samplings: Set[Feature]
+) -> pb.Sampling:
+    return pb.Sampling(
+        id=_identifier(src),
+        index=src.index(),
+        creator_event_id=(
+            _identifier(src.creator) if src not in input_samplings else None
+        ),
+    )
+
+
+def _unserialize_sampling(src: pb.Sampling) -> Sampling:
+    return Sampling(index=src.index, creator=None)
+
+
+def _type_to_proto(dtype) -> pb.Feature.DType:

```

Comment:
nit: naming of `_type_to_proto`, ..., `_attribute_from_proto` could follow the `_serialize`/`_unserialize` convention the other objects follow?

Reply:
*Done